VHACD does not compile with musl libc

<!-- Please search existing issues for potential duplicates before filing yours:
https://github.com/godotengine/godot/issues?q=is%3Aissue
-->

**Godot version:** 3bd49dab  
**OS:** Alpine Linux (Docker)

**Issue description:**
Godot Engine does not compile on Alpine (or other musl-based systems) due to the use of `PTHREAD_MUTEX_RECURSIVE_NP` in the third-party [VHACD](https://github.com/godotengine/godot/tree/master/thirdparty/vhacd) module. This appears to be a glibc-specific macro.

Here is the relevant log output:
```
[Initial build] Compiling ==> thirdparty/vhacd/src/VHACD.cpp
In file included from thirdparty/vhacd/inc/btScalar.h:190,
                 from thirdparty/vhacd/inc/btAlignedAllocator.h:23,
                 from thirdparty/vhacd/inc/btAlignedObjectArray.h:19,
                 from thirdparty/vhacd/inc/btConvexHullComputer.h:18,
                 from thirdparty/vhacd/src/VHACD.cpp:30:
thirdparty/vhacd/inc/vhacdMutex.h: In constructor 'VHACD::Mutex::Mutex()':
thirdparty/vhacd/inc/vhacdMutex.h:97:60: error: 'PTHREAD_MUTEX_RECURSIVE_NP' was not declared in this scope
         VHACD_VERIFY(pthread_mutexattr_settype(&mutexAttr, PTHREAD_MUTEX_RECURSIVE_NP) == 0);
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~
thirdparty/vhacd/inc/vhacdMutex.h:97:9: note: in expansion of macro 'VHACD_VERIFY'
         VHACD_VERIFY(pthread_mutexattr_settype(&mutexAttr, PTHREAD_MUTEX_RECURSIVE_NP) == 0);
         ^~~~~~~~~~~~
thirdparty/vhacd/inc/vhacdMutex.h:97:60: note: suggested alternative: 'PTHREAD_MUTEX_RECURSIVE'
         VHACD_VERIFY(pthread_mutexattr_settype(&mutexAttr, PTHREAD_MUTEX_RECURSIVE_NP) == 0);
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~
thirdparty/vhacd/inc/vhacdMutex.h:97:9: note: in expansion of macro 'VHACD_VERIFY'
         VHACD_VERIFY(pthread_mutexattr_settype(&mutexAttr, PTHREAD_MUTEX_RECURSIVE_NP) == 0);
         ^~~~~~~~~~~~
scons: *** [thirdparty/vhacd/src/VHACD.x11.tools.64.o] Error 1
scons: building terminated because of errors.

```

Something similar happens when cross-compiling from linux to windows with mingw.

```
thirdparty/vhacd/src/VHACD-ASYNC.cpp:307:7: error: 'thread' in namespace 'std' does not name a type
  307 |  std::thread      *mThread{ nullptr };
      |       ^~~~~~
thirdparty/vhacd/src/VHACD-ASYNC.cpp:10:1: note: 'std::thread' is defined in header '<thread>'; did you forget to '#include <thread>'?
    9 | #include <float.h>
  +++ |+#include <thread>
   10 | 
thirdparty/vhacd/src/VHACD-ASYNC.cpp:315:15: error: 'mutex' in namespace 'std' does not name a type
  315 |  mutable std::mutex      mMessageMutex;
      |               ^~~~~
thirdparty/vhacd/src/VHACD-ASYNC.cpp:10:1: note: 'std::mutex' is defined in header '<mutex>'; did you forget to '#include <mutex>'?
    9 | #include <float.h>
  +++ |+#include <mutex>
   10 | 
[ 50%] thirdparty/vhacd/src/VHACD-ASYNC.cpp: In member function 'virtual bool VHACD::MyHACD_API::Compute(const double*, uint32_t, const uint32_t*, uint32_t, const VHACD::IVHACD::Parameters&)':
thirdparty/vhacd/src/VHACD-ASYNC.cpp:53:3: error: 'mThread' was not declared in this scope; did you mean 'fread'?
   53 |   mThread = new std::thread([this, countPoints, countTriangles, _desc]()
      |   ^~~~~~~
      |   fread
thirdparty/vhacd/src/VHACD-ASYNC.cpp:53:17: error: expected type-specifier
   53 |   mThread = new std::thread([this, countPoints, countTriangles, _desc]()
      |                 ^~~
thirdparty/vhacd/src/VHACD-ASYNC.cpp:57:4: error: expected primary-expression before ')' token
   57 |   });
      |    ^
thirdparty/vhacd/src/VHACD-ASYNC.cpp: In member function 'virtual void VHACD::MyHACD_API::Cancel()':
thirdparty/vhacd/src/VHACD-ASYNC.cpp:166:7: error: 'mThread' was not declared in this scope; did you mean 'fread'?
  166 |   if (mThread)
      |       ^~~~~~~
      |       fread
thirdparty/vhacd/src/VHACD-ASYNC.cpp:169:11: error: type '<type error>' argument given to 'delete', expected pointer
  169 |    delete mThread;
      |           ^~~~~~~
thirdparty/vhacd/src/VHACD-ASYNC.cpp: In member function 'virtual void VHACD::MyHACD_API::Update(double, double, double, const char*, const char*)':
thirdparty/vhacd/src/VHACD-ASYNC.cpp:235:3: error: 'mMessageMutex' was not declared in this scope; did you mean 'mMessage'?
  235 |   mMessageMutex.lock();
      |   ^~~~~~~~~~~~~
      |   mMessage
thirdparty/vhacd/src/VHACD-ASYNC.cpp: In member function 'virtual void VHACD::MyHACD_API::Log(const char*)':
thirdparty/vhacd/src/VHACD-ASYNC.cpp:247:3: error: 'mMessageMutex' was not declared in this scope; did you mean 'mMessage'?
  247 |   mMessageMutex.lock();
      |   ^~~~~~~~~~~~~
      |   mMessage
thirdparty/vhacd/src/VHACD-ASYNC.cpp: In member function 'void VHACD::MyHACD_API::processPendingMessages() const':
thirdparty/vhacd/src/VHACD-ASYNC.cpp:267:4: error: 'mMessageMutex' was not declared in this scope; did you mean 'mMessage'?
  267 |    mMessageMutex.lock();
      |    ^~~~~~~~~~~~~
      |    mMessage
thirdparty/vhacd/src/VHACD-ASYNC.cpp:275:4: error: 'mMessageMutex' was not declared in this scope; did you mean 'mMessage'?
  275 |    mMessageMutex.lock();
      |    ^~~~~~~~~~~~~
      |    mMessage
```

--- a/thirdparty/vhacd/0003-fix-musl-build.patch
+++ b/thirdparty/vhacd/0003-fix-musl-build.patch
@@ -1,0 +1,15 @@
+diff --git a/thirdparty/vhacd/inc/vhacdMutex.h b/thirdparty/vhacd/inc/vhacdMutex.h
+index 6b09259200..d587dd6387 100644
+--- a/thirdparty/vhacd/inc/vhacdMutex.h
++++ b/thirdparty/vhacd/inc/vhacdMutex.h
+@@ -71,7 +71,9 @@
+ #include <pthread.h>
+ #endif
+ 
+-#if defined(__APPLE__)
++// -- GODOT start --
++#if defined(__APPLE__) || !defined(__GLIBC__)
++// -- GODOT end --
+ #define PTHREAD_MUTEX_RECURSIVE_NP PTHREAD_MUTEX_RECURSIVE
+ #endif
+ 

--- a/thirdparty/vhacd/inc/vhacdMutex.h
+++ b/thirdparty/vhacd/inc/vhacdMutex.h
@@ -71,7 +71,9 @@
 #include <pthread.h>
 #endif
 
-#if defined(__APPLE__)
+// -- GODOT start --
+#if defined(__APPLE__) || !defined(__GLIBC__)
+// -- GODOT end --
 #define PTHREAD_MUTEX_RECURSIVE_NP PTHREAD_MUTEX_RECURSIVE
 #endif
 

